lib/ansible/modules/network/nxos/nxos_acl_interface.py

```diff
@@ -85,83 +85,30 @@
 from ansible.module_utils.basic import AnsibleModule
 
 
-NAME = r'.*IP?\s+access list\s+(?P<name>\S+).*'
-INTERFACE = r'.*\s+(?P<interface>\w+(\d+)?\/?(\d+)?)\s-\s(?P<direction>\w+)\s+\W(?P<acl_type>\w+\s\w+)\W.*'
-
-
-def get_acl_interface(module, acl):
-    command = ['show ip access-list summary']
-    acl_list = []
-
+def check_for_acl_int_present(module, name, intf, direction):
+    command = ['show running-config aclmgr']
     body = run_commands(module, command)
-    body_split = body[0].split('Active on interfaces:')
-
-    for each_acl in body_split:
-        temp = {}
-        try:
-            match_name = re.match(NAME, each_acl, re.DOTALL)
-            name_dict = match_name.groupdict()
-            name = name_dict['name']
-        except AttributeError:
-            name = ''
-
-        temp['interfaces'] = []
-        for line in each_acl.split('\n'):
-            try:
-                match_interface = re.match(INTERFACE, line, re.DOTALL)
-                interface_dict = match_interface.groupdict()
-                interface = interface_dict['interface']
-                direction = interface_dict['direction']
-                acl_type = interface_dict['acl_type']
-            except AttributeError:
-                interface = ''
-                direction = ''
-                acl_type = ''
-
-            intf_temp = {}
-            if interface:
-                intf_temp['interface'] = interface
-            if acl_type:
-                intf_temp['acl_type'] = acl_type
-            if direction:
-                intf_temp['direction'] = direction
-            if intf_temp:
-                temp['interfaces'].append(intf_temp)
-        if name:
-            temp['name'] = name
-
-        if temp:
-            acl_list.append(temp)
-
-    existing_no_null = []
-    for each in acl_list:
-        if each.get('name') == acl:
-            interfaces = each.get('interfaces')
-            for interface in interfaces:
-                new_temp = {}
-                new_temp['name'] = acl
-                new_temp.update(interface)
-                existing_no_null.append(new_temp)
-    return existing_no_null
-
-
-def other_existing_acl(get_existing, interface, direction):
-    # now we'll just get the interface in question
-    # needs to be a list since same acl could be applied in both dirs
-    acls_interface = []
-    this = {}
-
-    if get_existing:
-        for each in get_existing:
-            if each.get('interface').lower() == interface:
-                acls_interface.append(each)
-
-        if acls_interface:
-            for each in acls_interface:
-                if each.get('direction') == direction:
-                    this = each
-
-    return acls_interface, this
+    body = str(body[0])
+
+    seq = re.compile(r"^interface " + re.escape(intf) + r"\n  ip access-group (.*) (.*)", re.M | re.I)
+    existing = False
+    while True:
```

Comment:
The output we get from the show command can be the following if both directions are configured.
interface Ethernet1/2
  ip access-group ANSIBLE_ACL in
  ip access-group ANSIBLE_ACL out

or only one of them is configured (egress)
interface Ethernet1/2
  ip access-group ANSIBLE_ACL out

or (ingress)
interface Ethernet1/2
  ip access-group ANSIBLE_ACL in
 
So depending upon the direction we are trying to match, we have to get rid of the extra line (for ex. remove the ingress if we are looking for egress) if they exist, so the loop is needed. I wrote the code without loop but the tests failed when I had both configured and so fixed it this way.